File: Iteration 1/code/CSE3063GRP12/src/core/models/concretes/CourseSession.java
package core.models.concretes;

import java.util.ArrayList;
import java.util.Map;

import core.enums.CourseDay;
import core.enums.CourseSlot;

public class CourseSession {
    private String id;
    private Map<CourseDay, ArrayList<CourseSlot>> courseSessions;
    private Lecturer lecturer;
    private String name;

    // Constructor with parameters
    public CourseSession(String id, Map<CourseDay, ArrayList<CourseSlot>> courseSessions, Lecturer lecturer) {
        this.id = id;
        this.courseSessions = courseSessions;
        this.lecturer = lecturer;

    }

    public Map<CourseDay, ArrayList<CourseSlot>> getCourseSessions() {
        return courseSessions;
    }


    public String getName() {
        return name;
    }

    public Lecturer getLecturer() {
        return lecturer;
    }

    public String getId() {
        return id;
    }

    public void setCourseSessions(Map<CourseDay, ArrayList<CourseSlot>> courseSessions) {
        this.courseSessions = courseSessions;
    }

    public void setLecturer(Lecturer lecturer) {
        this.lecturer = lecturer;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setId(String id) {
        this.id = id;

    }

    // Setters

}

File: Iteration3/code/CSE3063GRP12/src/core/models/concretes/CourseSession.py


class CourseSession:
    def __init__(self, id, courseSessions, lecturer):
        self.id = id
        self.courseSessions = courseSessions  # dictionary mapping CourseDay to a list of CourseSlot
        from core.models.concretes.Lecturer import Lecturer
        self.lecturer = Lecturer(**lecturer) 
        self.name = None
    
    def to_dict(self):
        return {
            "id": self.id,
            "courseSessions": {courseDay.value: [courseSlot.to_dict() if hasattr(courseSlot, 'to_dict') else courseSlot for courseSlot in courseSlots] for courseDay, courseSlots in self.courseSessions.items()},
            "lecturer": self.lecturer.to_dict() if hasattr(self.lecturer, 'to_dict') else self.lecturer,
            "name": self.name
        }

